ext/drreg/drreg.c

```diff
@@ -1292,7 +1310,9 @@ drreg_reservation_info(void *drcontext, reg_id_t reg, opnd_t *opnd OUT,
     };
     per_thread_t *pt = get_tls_data(drcontext);
     drreg_status_t res;
-    if (reg < DR_REG_START_GPR || reg > DR_REG_STOP_GPR || !pt->reg[GPR_IDX(reg)].in_use)
+    if (!(reg >= DR_REG_START_GPR && reg <= DR_REG_STOP_GPR &&
+          pt->reg[GPR_IDX(reg)].in_use) &&
+        !(reg == DR_REG_NULL && pt->aflags.in_use))
```

Comment:
Please update the docs, which today only say that DR_REG_NULL for flags can be passed to the _ex version, not the base version.

Reply:
Also, if this now matches the _ex version, this check can be removed and the check in _ex relied upon.

Reply:
> Please update the docs

Done

> Also, if this now matches the _ex version, this check can be removed and the check in _ex relied upon.

The _ex version has a different check. Unlike this one, it doesn't seem to care whether the queried reg is actually reserved or not. So keeping this check separate.